### PR TITLE
basic redux

### PR DESCRIPTION
**Pull request by Alphy11**

### CODE REVIEW

**Review by vermentor — APPROVED** [review]

File: server/src/gameMachine/index.ts [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+        case "KickPlayer": {
+          const { [action.player]: kickedPlayerNeighbor, ...nextState } = state;
+          const oprhanedNeighbor = toEntries(nextState).find(
+            ([_, value]) => value?.rightNeighbor === action.player,
+          );
+
+          if (oprhanedNeighbor) {
+            return {
+              ...nextState,
+              [oprhanedNeighbor[0]]: {
```
Comment (vermentor) [inline server/src/gameMachine/index.ts:69]: Very nitpicky, but I'd prefer orphaned neighbor was desctructed from the get.
Also typo "oprhaned"

**Comment by Alphy11** [comment]
Can’t be. It might be null
________________________________

File: server/src/gameMachine/index.ts [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+            ...state,
+            [action.player]: action.newAlignment,
+          };
+        default:
+          return state;
+      }
+    },
+
+    gmSecretHash: (state) => state ?? "t",
+    gameStatus: (state = "PlayersJoining", action) => {
```
Comment (vermentor) [inline server/src/gameMachine/index.ts:150]: We should be careful that having just two statuses doesn't break anything in the interim we work on statuses

**Review by vermentor — APPROVED** [review]

File: server/src/gameMachine/index.ts (hunk 1 above)
Comment (Alphy11) [reply server/src/gameMachine/index.ts]: Can't it might be null

**Pull request merged**